pkg/types/interface.go

```diff
@@ -33,10 +40,21 @@ type NamespaceInterface interface {
 	SleepNamespace(ctx context.Context, namespace string) error
 }
 
+// PreviewInterface represents the client that connects to the preview functions
 type PreviewInterface interface {
 	List(ctx context.Context) ([]Preview, error)
 }
 
+// PipelineInterface represents the client that connects to the pipeline functions
+type PipelineInterface interface {
+	DeployPipeline(ctx context.Context, name, repository, branch, filename string, variables []Variable) (*GitDeployResponse, error)
```

Comment:
As this is now within a PipelineInterface, maybe it can be called just `Deploy`. The same for `Destroy`, `GetResourcesStatus` and `GetByName`.

If it implies too much refactor, no worries, it is ok as it is